packages/terra-table/src/subcomponents/ColumnHeader.jsx

```diff
@@ -141,13 +141,15 @@ const ColumnHeader = (props) => {
         className={cx('column-header-row', { hidden: !hasVisibleColumnHeaders })}
         height={hasVisibleColumnHeaders ? headerHeight : undefined}
       >
-        {columns.map((column, columnIndex) => (
+
+        {columns.filter((column) => (column.columnSpan > 1 || !(column.columnSpanIndex))).map((column, columnIndex) => (
```

Comment:
I would not expect the column header component to require changes.  We would continue to use our column shape information.  There should not be a need to filter the data.

Reply:
Updated.